### PR TITLE
refactor: change './bin' to 'LOCALBIN'

### LINKED ISSUE #6637: Change `./bin` to `LOCALBIN`
#### Describe the problem you need a feature to resolve.

Path to some tools binaries is hard coded with `./bin`, instead of using `LOCALBIN` (https://github.com/kubernetes-sigs/kubebuilder/blob/3c9cf656ce4d62a10c8e78a1c0f066de47c983db/pkg/plugins/golang/v3/scaffolds/internal/templates/makefile.go#L185).

Example
https://github.com/operator-framework/operator-sdk/blob/e001847a67f16d7173ae4fee7cff57cd4405edbb/internal/plugins/manifests/v2/init.go#L207

#### Describe the solution you'd like.

It would be also nice if tools binaries variables (`OPERATOR_SDK`, `OPM`, `OPERATOR_SDK_VERSION` and `OPM_VERSION`, which does not exist) could be grouped in here https://github.com/kubernetes-sigs/kubebuilder/blob/3c9cf656ce4d62a10c8e78a1c0f066de47c983db/pkg/plugins/golang/v3/scaffolds/internal/templates/makefile.go#L189-L196

### PR DESCRIPTION
**Pull request by lunarwhite**
**Description of the change:**

avoid hardcoded tool binary variables `OPERATOR_SDK` and `OPM`.

**Motivation for the change:**

fixes #6637

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

File: internal/plugins/manifests/v2/init.go
Comment (everettraven) [inline internal/plugins/manifests/v2/init.go:188]: Does LOCALBIN already exist in the makefile that we are going to be adding this information to?

**Review by everettraven — CHANGES_REQUESTED** [review]
Thanks for creating this PR @lunarwhite ! Would you mind regenerating the testdata with `make generate` and pushing up the changes?

File: internal/plugins/manifests/v2/init.go
Comment (lunarwhite) [reply internal/plugins/manifests/v2/init.go:188]: Good catch! I'll update.

**Comment by lunarwhite** [comment]
Hi @everettraven, thanks for reviewing. At this time, `make generate` will also auto change Copyright `2023` to `2024`. So I separate logic into two commits, to make reviewing process easier.

Adding label to squash commits:

/label tide/merge-method-squash

**Comment by lunarwhite** [comment]
Hi @everettraven, it seems that workflow jobs require maintainer's approval.

**Comment by lunarwhite** [comment]
Last commit will fix "sanity / docs (pull_request) Failing after 2m"
```
Running ["ScriptCheck", "LinkCheck", "ImageCheck"] on ["/target"] on *.html... 


Checking 777 external links...
- /target/docs/best-practices/observability-best-practices/index.html
  *  External link https://github.com/operator-framework/operator-sdk/tree/master/testdata/go/v4-alpha/monitoring/memcached-operator/monitoring failed: 404 No error
Ran on 170 files!

HTML-Proofer found 1 failure!


sdk-html
exiting @ Thu Jan 11 19:56:04 UTC 2024
make: *** [Makefile:156: test-docs] Error 1
Error: Process completed with exit code 2.
```

It's a little weird as I didn't touch this kind of file. Anyway, it failed mainly because https://github.com/operator-framework/operator-sdk/blob/c557c5410472dcee533f09995f7a6c35ab01fe30/changelog/generated/v1.33.0.md?plain=1#L6

Now I believe it's ok to re-run jobs, sorry for the frequent requests.

**Pull request merged**
